### PR TITLE
Feature/restrict function pointer by name [depends-on: #5168]

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
This works similar to restrict-function-pointer, but for names of individual
function pointer variables (globals, locals, parameters) rather than call sites.
This isn't applicable to all situations (for example, calling function pointers
in structs or function pointers returned from functions), but is more readily
applicable to some common use scenarios (e.g. global function pointers loaded at
start time like in OpenGL).

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/cbmc/cbmc_parse_options.cpp [hunk 1]
```diff
@@ -11,8 +11,8 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "cbmc_parse_options.h"
 
-#include <fstream>
 #include <cstdlib> // exit()
+#include <fstream>
```
Comment (xbauch) [inline src/cbmc/cbmc_parse_options.cpp:15]: It looks like you called `clang-format` on the whole file: maybe a separate commit?

**Review by xbauch — APPROVED** [review]
Looks good.

File: src/cbmc/cbmc_parse_options.cpp (hunk 1 above)
Comment (hannes-steffenhagen-diffblue) [reply src/cbmc/cbmc_parse_options.cpp:15]: Nah I'll undo it.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: ee5b42b6f7266c8c8aec38e52bea7379bc00ceac).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/135430669

**Review by karkhaz — APPROVED** [review]
This seems like a more convenient way of using this feature than indexing the calls in a function––thanks!

File: src/util/irep.h [hunk 2]
```diff
@@ -50,6 +50,11 @@ inline const std::string &id2string(const irep_idt &d)
   #endif
 }
 
+inline irep_idt string2id(const std::string &id_string)
+{
+  return irep_idt{id_string};
+}
+
```
Comment (tautschnig) [inline src/util/irep.h:57]: This seems unnecessary (and unused)?

**Pull request closed without merging**